lib-core/src/main/java/com/kinandcarta/create/proxytoggle/lib/core/extensions/LiveDataExtensions.kt

```diff
@@ -1,4 +1,4 @@
-package com.kinandcarta.create.proxytoggle.manager.extensions
+package com.kinandcarta.create.proxytoggle.lib.core.extensions
```

Comment:
what is lib core for? is it to be used by presentation and view layer only?

Reply:
It's just a way to abstract all common and shared functionalities that will be used by different modules. I don't like the name "base" 😬

Reply:
if it's to be used by all modules then you are forcing all modules to depend on live data and android. The network, domain layer etc should be aware of LiveData. This breaks what clean architecture is and could lead to a less experienced developer trying to use dependencies were they shouldn't be used

Reply:
I usually had a common package for each layer. And a general pure Kotlin common package for the entire app

Reply:
Fair point. Although, the app won't have a domain or  network layer. Actually it won't grow much more than a file or two, it's a very simple app for just setting a proxy!
If I do this, I'll create a new module for just one file or two... Do you think it's worth it?

Reply:
If it is that small then what is the point of this lib core? seems pointless
you can just place the classes in a package in a layer they a relevant to maybe

Reply:
sorry, just have been trying to understand what this lib core is, as I have seen it in other projects and it didn't make sense to me what it is

Reply:
Agreed - `lib-core` became kind of a dumping ground where we'd stick everything we need in all other modules - probably clearly defining multiple libs for the core needs of each layer makes more sense.

But for this tiny app.. why do we even have multiple modules? 😄 

Feel free to just chuck everything in the app module, probably it's overkill to have more than one module here?

Reply:
Well I wanted to have two separate feature modules: one for the proxy management (inside the app) and another -separated from the other- for the home screen widget. Since they share some classes (eg: models) they need a "core" module that has the shared functionality

Reply:
Agreed - `lib-core` became kind of a dumping ground where we'd stick everything we need in all other modules - probably clearly defining multiple libs for the core needs of each layer makes more sense.

But for this tiny app.. why do we even have multiple modules? 😄 

Feel free to just chuck everything in the app module, probably it's overkill to have more than one module here?

Reply:
> sorry, just have been trying to understand what this lib core is, as I have seen it in other projects and it didn't make sense to me what it is

Agreed - `lib-core` became kind of a dumping ground where we'd stick everything we need in all other modules - probably clearly defining multiple libs for the core needs of each layer makes more sense.

But for this tiny app.. why do we even have multiple modules? 😄 

Feel free to just chuck everything in the app module, probably it's overkill to have more than one module here?

Reply:
Sure 🤷 if you guys is better not having modules

Reply:
Done, everything in one gigantic module now!